tests/tests/bitasset_tests.cpp

```diff
@@ -877,113 +737,41 @@ BOOST_AUTO_TEST_CASE( bitasset_evaluator_test_after_922_931 )
    // back by self
    BOOST_TEST_MESSAGE( "Back by itself" );
    op.new_options.short_backing_asset = bit_usd_id;
-   try
-   {
-      evaluator.evaluate(op);
-      BOOST_FAIL("Backing by self should have failed.");
-   }
-   catch (fc::exception &ex)
-   {
-      BOOST_CHECK_EQUAL( ex.to_string(log_level_all),
-            "Assert Exception: op.new_options.short_backing_asset != asset_obj.get_id(): Cannot update an asset to be backed by itself." );
-   }
+   REQUIRE_EXCEPTION_WITH_TEXT(evaluator.evaluate(op), "Cannot update an asset to be backed by itself");
    op.new_options.short_backing_asset = correct_asset_id;
 
    // prediction market with different precision
    BOOST_TEST_MESSAGE( "Prediction market with different precision" );
-   op.asset_to_update = prediction->get_id();
-   op.issuer = prediction->issuer;
-   try
-   {
-      evaluator.evaluate(op);
-      BOOST_FAIL("precision difference should have failed.");
-   }
-   catch (fc::exception &ex)
-   {
-      BOOST_CHECK_EQUAL( ex.to_string(log_level_all),
-            "Assert Exception: asset_obj.precision == new_backing_asset.precision: The precision of the asset and backing asset must be equal." );
-   }
+   op.asset_to_update = asset_objs.prediction->get_id();
+   op.issuer = asset_objs.prediction->issuer;
+   REQUIRE_EXCEPTION_WITH_TEXT(evaluator.evaluate(op), "The precision of the asset and backing asset must");
    op.asset_to_update = bit_usd_id;
-   op.issuer = bit_usd.issuer;
+   op.issuer = asset_objs.bit_usd->issuer;
 
    // checking old backing asset instead of new backing asset
    BOOST_TEST_MESSAGE( "Correctly checking new backing asset rather than old backing asset" );
-   op.new_options.short_backing_asset = sixbit->get_id();
-   try
-   {
-      evaluator.evaluate(op);
-      BOOST_FAIL("Changing committee controlled MPA to be backed by UIA should have failed.");
-   }
-   catch (fc::exception &ex)
-   {
-      BOOST_CHECK_EQUAL( ex.to_string(log_level_all),
-            "Assert Exception: new_backing_asset.get_id() == asset_id_type(): May not modify a blockchain-controlled market asset to be backed by an asset which is not market issued asset nor CORE." );
-   }
-   op.new_options.short_backing_asset = prediction->get_id();
-   try
-   {
-      evaluator.evaluate(op);
-      BOOST_FAIL("Changing committee controlled MPA to be backed by a MPA which is backed by UIA should have failed.");
-   }
-   catch (fc::exception &ex)
-   {
-      BOOST_CHECK_EQUAL( ex.to_string(log_level_all),
-            "Assert Exception: new_backing_asset.bitasset_data(d).options.short_backing_asset == asset_id_type(): May not modify a blockchain-controlled market asset to be backed by an asset which is not backed by CORE." );
-   }
+   op.new_options.short_backing_asset = asset_objs.bit_usdbacked->get_id();
```

Comment:
According to the original message, should try to change backing asset to an UIA. Also the message below should indicate this. My code looked buggy but actually was not buggy, since `sixbit` was actually not a bitAsset, although it's better to use `BLAH` to be more obvious.